spec/Model.spec.js

```diff
@@ -411,6 +411,73 @@ describe('Model', () => {
       expect(todo.dirtyAttributes).toHaveLength(1)
       expect(todo.dirtyAttributes[0]).toEqual('options.variety')
     })
+
+    it('tracks removed to relationships', async () => {
+      const todo = store.add('organizations', { id: 11, title: 'Buy Milk' })
+      const note = store.add('notes', {
+        id: 11,
+        description: 'Example description'
+      })
+
+      todo.notes.add(note)
+      todo.setPreviousSnapshot()
+      expect(todo.dirtyAttributes).toEqual([])
+      todo.notes.remove(note)
+      expect(todo.dirtyAttributes).toEqual(['relationships.notes.0'])
+    })
+
+    it('tracks added relationship', async () => {
+      const todo = store.add('organizations', { id: 11, title: 'Buy Milk' })
+      const note = store.add('notes', {
+        id: 11,
+        description: 'Example description'
+      })
+
+      expect(todo.dirtyAttributes).toEqual([])
+      todo.notes.add(note)
+      expect(todo.dirtyAttributes).toEqual(['relationships.notes.0'])
+    })
+
+    it('reverts to empty after adding and then removing a relationship', async () => {
+      const todo = store.add('organizations', { id: 11, title: 'Buy Milk' })
+      const note = store.add('notes', {
+        id: 11,
+        description: 'Example description'
+      })
+
+      expect(todo.dirtyAttributes).toEqual([])
+      todo.notes.add(note)
+      expect(todo.dirtyAttributes).toEqual(['relationships.notes.0'])
```

Comment:
What is the structure of this key mean?

Reply:
`record.relationships.notes.0` is the relationship that has changed since the last snapshot. In other words, the object that you would access at `record.relationships.notes[0]` has changed.